app/controllers/v0/addresses_controller.rb

```diff
@@ -2,39 +2,39 @@
 module V0
   class AddressesController < ApplicationController
     def show
-      response = service.get_address(@current_user)
+      response = service.get_address
       render json: response,
              serializer: AddressSerializer
     end
 
     def update
       address = EVSS::PCIUAddress::Address.build_address(params)
       raise Common::Exceptions::ValidationErrors, address unless address.valid?
-      response = service.update_address(@current_user, address)
+      response = service.update_address(address)
       render json: response,
              serializer: AddressSerializer
     end
 
     def countries
-      response = strategy.cache_or_service(:countries) { service.get_countries(@current_user) }
+      response = strategy.cache_or_service(:countries) { service.get_countries }
       render json: response,
              serializer: CountriesSerializer
     end
 
     def states
-      response = strategy.cache_or_service(:states) { service.get_states(@current_user) }
+      response = strategy.cache_or_service(:states) { service.get_states }
       render json: response,
              serializer: StatesSerializer
     end
 
     private
 
     def service
-      @service ||= EVSS::PCIUAddress::Service.new
+      EVSS::PCIUAddress::Service.new(@current_user)
```

Comment:
Minor yak shaving - not sure how the memoization here would have protected against anything as far as the request goes since, as you mentioned, a new instance of the controller is used for each request. If we'd left this then the service would have just been memoized for multiple calls within the same action. Prefer explicitly not memoizing it though.